### PR TITLE
Add using WPILIB's PIDController article

### PR DESCRIPTION
**Pull request by tomiam8**
In the article, there are sections:
* An intro
* Advantages & Disadvantages of the WPILIB PID Controller Class
* Guide to implementing the Controller
* List and explanation of the various options & functions in the PIDController
* Guide on using a Velocity PID Control
* Section on using PID Subsystems
* Explanation of the various PID classes in WPILIB (may not include all of them though)
* Section on Adding Ramping for Motors

I also added a feed-forward section in the PID Theory article

### CODE REVIEW

File: control/using_WPILIB's_pid_controller.rst [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+Pre-exisiting tested nature
+	The PID Controller class has alread been written, tested and debugged. Doing it your self dosen't provide that certainty, and will near definitely take longer.
+
+	
+Disadvantges of the WPILIB PID Controller Class
+-------------------------------------------------
+
+Could be better control-theory wise
+	While the WPILIB PID Controller class is quite good, for advance teams, it's lack of an acceleration term for example, means that it could be improved upon. But, if this is why you don't want to use it, then you probably don't need this article.
```
Comment (Twinters007) [inline control/using_WPILIB's_pid_controller.rst]: its not it's

File: control/using_WPILIB's_pid_controller.rst [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+Potentially slow(er) on the RIO
+	As mentioned earlier, the PID Controller class must run on the RoboRIO. Other alternatives, like using the Talon SRX's inbuilt PID controller, will operate faster and so with better control.
+
+Ramping motorsup
```
Comment (Twinters007) [inline control/using_WPILIB's_pid_controller.rst]: space between motors and up

File: control/using_WPILIB's_pid_controller.rst [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	The PID Controller class can result in very jerky motion which can be bad for geraboxes. However, increasing the D term should smoothen out sudden changes. Alternatively, see "Adding Ramping for Motors".
```
Comment (Twinters007) [inline control/using_WPILIB's_pid_controller.rst]: smooth out sudden

File: control/using_WPILIB's_pid_controller.rst [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+	
+Linear output assumption
+	The PID Controller, like nearly all versions of PID, assumes a linear output, however in reality motor responses are curves.
+
+	
+Implementing a basic PID Control
+------------------------------------
+PIDController constuctors:
```
Comment (Twinters007) [inline control/using_WPILIB's_pid_controller.rst]: I'd rather not just copy/paste the documentation here. I suggest a link to the docs.

File: control/using_WPILIB's_pid_controller.rst [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+====================  =============================================================================================================================================================================================================================
+
+
+Velocity PID Control
+---------------------
+To use PID Controller to maintain a velocity - say for a shooter fly wheel or closed loop driving:
+
++ You should use a feedforward term (Kf)
++ Your PIDSource should probably have a PIDSourceType of kRate
++ Be carefull of what your PIDSource is giving - for example, if you use an encoder, and it gives encoder positions, but you want speed, then you might need to wrap it with your own code that gives the rate of change instead.
```
Comment (Twinters007) [inline control/using_WPILIB's_pid_controller.rst]: careful

File: control/using_WPILIB's_pid_controller.rst [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+
+Using PID Subsystem
+------------------------
+WPILIB provides the PID Subsytem class to provide convenience methods to run a PIDController on a subsytem for simple cases. For example, if you had an elevator subsytem that needed to stay at the same height, you could use a PIDSubsystem for that.
```
Comment (Twinters007) [inline control/using_WPILIB's_pid_controller.rst]: WPILib

File: control/using_WPILIB's_pid_controller.rst [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+To use, rather than extending Subsystem, extend PIDSubsytem.
+
+You will need to define the functions returnPIDInput and usePIDOutput to give to the PIDController, and you will want to in the constructor for your subsytem call super(p, i, d, f, period)
```
Comment (Twinters007) [inline control/using_WPILIB's_pid_controller.rst]: wrap super in a codeblock. Also, I think this should have either an example or a link to the example in the examples repo of WPILib

File: control/using_WPILIB's_pid_controller.rst [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+
+You can access the internal PIDController with getPIDController()
+
+
+Explanation of the various PID WPILIB class's
```
Comment (Twinters007) [inline control/using_WPILIB's_pid_controller.rst]: WPILib

File: control/using_WPILIB's_pid_controller.rst [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
++-------------------+-----------------------------------------------------------------------------------------------------------------------------------------------+
+
+Adding Ramping for motors
+---------------------------
+As mentioned earlier, the best way is generally to increase your D term as it will smooth out sudden changes.
+However, alternative options, if for some reason you could not change your D term:
+
++ Create a wrapper function for PIDWrite that dampens motors. This function would store the previous output to the motor, and if given a new output that was say greater than 0.2 higher, it would only increase it by 0.2, and then increase it more after a brief wait. Note that this will reduce the effictiveness of your control, and will most likely mess up the I term of the PID loop
++ Dynamically change the minimum / maximum values of your PID Controller. Say, whever PIDWrite get's called, change the PIDController's maximum and minimum values to be around a certain band. This is basically the first option, but a bit better as it will limit the I term and stop it from going crazy.
```
Comment (Twinters007) [inline control/using_WPILIB's_pid_controller.rst]: Showing is better than telling. Let's see some examples of what this looks like

**Comment by tomiam8** [comment]
@Twinters007 I've added/changed the stuff you asked for

**Pull request merged**